src/helperFunctions/docker.py

```diff
@@ -1,48 +1,62 @@
 import logging
 from contextlib import suppress
-from typing import Optional, Tuple
 
 import docker
-from docker.errors import DockerException
-from docker.types import Mount
+from docker.errors import APIError, DockerException, ImageNotFound
 from requests.exceptions import ReadTimeout
 
 
-def run_docker_container(  # pylint: disable=too-many-arguments
-    image: str, timeout: int = 300, command: Optional[str] = None, reraise: bool = False, privileged: bool = False,
-    mount: Optional[Tuple[str, str]] = None, label: str = 'Docker', include_stderr: bool = True
-) -> str:
-    '''
-    Run a docker container and get its output.
-
-    :param image: the name of the docker image
-    :param timeout: a timeout after which the execution is canceled
-    :param command: the command to run in the container (optional)
-    :param reraise: re-raise exceptions if they occur (timeout and docker exceptions)
-    :param privileged: Run container with elevated privileges.
-    :param mount: specifies a directory that gets mounted into the container;
-                  structure: `(path_inside_container, source_path)`
-    :param label: label used for logging output
-    :param include_stderr: include stderr of the container in the output
-    :return: the output of the docker container
-    '''
-    container = None
+def run_docker_container(image: str, logging_label: str = 'Docker', timeout: int = 300,  stderr=True, **kwargs):
```

Comment:
> abstract from the docker API so that you don't have read the docs and can instead rely on the docstring

It might just be my opinion, but I think the function is more usable now.
When you have to read our docstring to use the function, why not read the one from dockerpy?

Hmm, what benefit will the `NamedTuple` have?

```python
# Current code
output, exit_code = run_docker_container(...)

# Code that can be written with a NamedTuple
result = run_docker_container(...)
output = result.output
exit_code = result.exit_code

```

Reply:
It's actually more like:

```python
# Current code
output, exit_code = run_docker_container(...)
if exit_code != 0: ...

output, _ = run_docker_container(...)
return output

# Code that can be written with a NamedTuple
result = run_docker_container(...)
if result.exit_code != 0: ...

result = run_docker_container(...)
return result.output
```

but yes, there is no real benefit (I'm just not very fond of returning tuples in general because then you end up accessing stuff by index most of the time)

Reply:
> I'm just not very fond of returning tuples 

Me neither, maybe we should return a `subprocess.CompletedProcess` instance.

Reply:
That was actually what I thought about when I suggested the `NamedTuple` (because it behaves pretty similar from the callers view)

Reply:
Changed the code to return a `subprocess.CompletedProcess` instance.